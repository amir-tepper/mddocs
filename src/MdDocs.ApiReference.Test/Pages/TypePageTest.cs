using System.Linq;
using Grynwald.MdDocs.ApiReference.Configuration;
using Grynwald.MdDocs.ApiReference.Model;
using Grynwald.MdDocs.ApiReference.Model.XmlDocs;
using Grynwald.MdDocs.ApiReference.Pages;
using Microsoft.Extensions.Logging.Abstractions;

namespace Grynwald.MdDocs.ApiReference.Test.Pages
{
    /// <summary>
    /// Tests for <see cref="TypePage"/>
    /// </summary>
    public class TypePageTest : PageTestBase<TypeDocumentation, TypePage>
    {

        protected override TypeDocumentation CreateSampleModel()
        {
            var assembly = Compile(@"
                namespace MyNamespace
                {
                    public class Class1
                    { }
                }
            ");

            var assemblyDocumentation = new AssemblyDocumentation(assembly, NullXmlDocsProvider.Instance, NullLogger.Instance);
            return assemblyDocumentation.MainModuleDocumentation.Types.Single();
        }

        protected override TypePage CreatePage(TypeDocumentation model, ApiReferenceConfiguration configuration)
        {
            return new TypePage(NullLinkProvider.Instance, configuration, model, NullLogger.Instance);
        }
    }
}
