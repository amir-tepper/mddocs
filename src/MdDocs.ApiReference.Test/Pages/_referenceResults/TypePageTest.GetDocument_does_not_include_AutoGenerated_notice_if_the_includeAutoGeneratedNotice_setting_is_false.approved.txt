# Class1 Class

**Namespace:** MyNamespace

**Assembly:** DynamicTestAssembly

```csharp
public class Class1
```

**Inheritance:** object → Class1

## Constructors

| Name     | Description |
| -------- | ----------- |
| Class1() |             |

___

*Documentation generated by [MdDocs](https://github.com/ap0llo/mddocs)*
